### DISCUSSION TITLE
Subdomain / Main domain usage issue

### DISCUSSION THREAD

**Discussion by anebi** [postalserver/postal#1940] [Help with using Postal] [answered]
Hello,

I am trying to achieve Mailgun way of work with Postal.

In Mailgun I can add subdomain like 'mg.mydomain.tld' and then I can add all the required DNS records for that subdomain. Like this all emails are sent with sender 'mg.mydomain.tld' but in From header I can still use 'mydomain.tld' or different subdomains.

I try to achieve the same with Postal but it looks like it is not possible. In domains section I added postal.domain.tld, went through the verification process and then created SMTP credentials. Now I am trying to send emails with From header mydomain.tld but Postal doesn't allow me.

I did some research and I found someone saying that I need to set Sender header but unfortunatelly I don't have such options. 

To be honest I really don't understand why Postal doesn't work the way Mailkgun does, it is much better and gives big flexibility.

Right now what I am trying to achieve is to have 2 SMTP servers where I want to set subdomains like this:

trans.mydomain.tld
mkt.mydomain.tld

From header for both SMTPs must be mydomain.tld. The idea is to separate transactional emails from marketing campaigns. This will allow me to have much better control. Unfortunately it seems like it is not possible right now.

Is there some solution for this?

Thank you in advance!

Kind regards.
Ali Nebi

**Comment by willpower232** [comment, marked as answer]
Postal verifies the domain that is being used to try to ensure that the message will be delivered. You can add as many subdomains or top level domains as you want to Postal so the easiest thing to do would be to add mydomain.tld.

As you say, sender header allows you to use other domains but may not be as easy to set up.

**Comment by anebi** [comment]
Thank you very much for your reply.

I wasn't sure if there will be some conflict if I try to add the same domain to multiple SMTP servers. If Postal supports using same domain with multiple SMTP servers then this definitely solves the problem I have because I can always use mydomain.tld without going into trouble with subdomains usage. :)

**Reply by willpower232** [reply]
The only potential problem would be with the DKIM record which is unique in certain places. The best thing would be to add the domain to the organization and then that would let all servers under that organization use it.

**Reply by anebi** [reply]
Yeah. I'll run some tests today but in general what you suggest is expected to work fine. Thanks! :)
